compiler/ccgtypes.nim

```diff
@@ -18,6 +18,7 @@ type
   TypeDescKind = enum
     dkParam #skParam
     dkRefParam #param passed by ref when {.byref.} is used. Cpp only. C goes straight to dkParam and is handled as a regular pointer
+    dkRefGenericParam #param passed by ref when {.byref.} is used that is also a generic. Cpp only. C goes straight to dkParam and is handled as a regular pointer
```

Comment:
Why is this special case even necessary? In theory it is not.

Reply:
Because it would enter in 
```nim
    if kind == dkRefParam or kind == dkRefGenericParam and origTyp.kind == tyGenericInst:
      result.add("&")
```
 and produce `typedef Box<Foo&> Instanced;`  (notice `Foo&`) what we want is `Instanced&` in the function, the type should still be `typedef Box<Foo> Instanced`. Take in mind that it will enter there twice, once for `Instanced` and another for `Foo` hence the distinction.